### ISSUE TITLE
Commutative Elemwise Operations are not merged

### ISSUE THREAD

**Issue by ricardoV94** [aesara-devs/aesara#958]
```python
import aesara
import aesara.tensor as at

x, y = at.scalars("x", "y")
f = aesara.function([x, y], [x + y, x + y, y + x, y + x])
aesara.dprint(f)
```

```python
DeepCopyOp [id A] ''   1
 |Elemwise{add,no_inplace} [id B] ''   0
   |x [id C]
   |y [id D]
Elemwise{add,no_inplace} [id B] ''   0
DeepCopyOp [id E] ''   3
 |Elemwise{add,no_inplace} [id F] ''   2
   |y [id D]
   |x [id C]
Elemwise{add,no_inplace} [id F] ''   2
```

**Comment by brandonwillard** [comment]
This might require a simple/naive canonicalization that involves string name/auto-ID ordering of commutative terms.  Such a canonicalization should be consistent across recreations of equivalent graphs, so I'm not sure if a direct auto-ID approach is actually sound, though.

**Comment by brandonwillard** [comment]
> One more direct way would be to include op information about what inputs are commutative and in the merge optimization we compare those via set equality instead of order equality like we do now for all inputs.
> 
> Just some care would be needed for repeated inputs, so maybe via a Counter instead of set

Yes, that's basically how the permutations and AC unification work in `kanren`.
